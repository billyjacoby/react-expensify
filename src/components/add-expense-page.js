import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './expense-form';
import {addExpense} from '../actions/expenses';

const AddExpensePage = (props) => (
  <div>
    This is from my create page
    <ExpenseForm
      onSubmit={(expense) => {
        props.dispatch(addExpense(expense))
        // the following changes to page without reloading
        props.history.push('/');
      }}
    />
  </div>
);

export default connect()(AddExpensePage);
