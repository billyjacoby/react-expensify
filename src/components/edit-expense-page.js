import React from 'react';

export default (props) => {
  return <div>Editing expense id {props.match.params.id}</div>
}
