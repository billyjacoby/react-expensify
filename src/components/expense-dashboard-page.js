import React from 'react';

import ExpenseList from './expense-list';
import ExpenseListFilters from './expense-list-filters';
import ExpensesSummary from './expenses-summary';

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpensesSummary />
    <ExpenseList/>
  </div>
);

export default ExpenseDashboardPage;
