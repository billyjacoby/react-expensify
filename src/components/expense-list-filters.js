import React from 'react';
import {connect} from 'react-redux';
import { DateRangePicker } from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

import {setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from '../actions/filters';

class ExpenseListFilters  extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      calendarFocused: null
    }
  }

  onDatesChange = ({startDate, endDate}) => {
    this.props.dispatch(setStartDate(startDate));
    this.props.dispatch(setEndDate(endDate));
  }

  render () {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={(e) => {this.props.dispatch(setTextFilter(e.target.value));}}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={(e) => {e.target.value === "date" ? this.props.dispatch(sortByDate()) : this.props.dispatch(sortByAmount());}}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>

        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          // onDatesChange={({ startDate, endDate }) => this.onDatesChange({startDate, endDate})}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={(focusedInput) => this.setState({calendarFocused: focusedInput})}
        />
      </div>
    )
  }

}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
};

export default connect(mapStateToProps)(ExpenseListFilters);
