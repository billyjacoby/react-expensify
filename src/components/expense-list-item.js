import React from 'react';
import {connect} from 'react-redux';

import {removeExpense} from '../actions/expenses';

const ExpenseListItem = ({description, amount, createdAt, note, id, dispatch}) => (
  <div>
    <h3>{description}</h3>
    <p>{amount} - {createdAt}</p>
    {
      note && <p>Note: {props.expense.note}</p>
    }
    <button
      onClick = {() => dispatch(removeExpense({id}))}
    >Remove</button>
  </div>
);

export default connect()(ExpenseListItem);
