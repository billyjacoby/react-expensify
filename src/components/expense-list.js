import React from 'react';
import {connect} from 'react-redux';

import ExpenseListItem from './expense-list-item';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
  <div>
    {
      (
        props.expenses.map((expense) => {
          return (
            <ExpenseListItem {...expense} key={expense.id}/>
          )
        })
      )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  }
};

export default connect(mapStateToProps)(ExpenseList)
