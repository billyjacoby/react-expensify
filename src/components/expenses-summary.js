import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = (props) =>{
  const expenseWord = props.expenseCount === 1 ? "expense" : "expenses";
  return (
    <div>
      {
        props.expenseCount > 0 ?
        <div>
          <h3>You are viewing {props.expenseCount} {expenseWord} totaling {numeral(props.expensesTotal / 100).format('$0,0.00')}</h3>
        </div> :
        <div>
          <h3>No expenses found!</h3>
        </div>
      }

    </div>
  );
}

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }
};

export default connect(mapStateToProps)(ExpensesSummary);
