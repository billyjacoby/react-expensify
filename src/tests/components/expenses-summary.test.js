import React from 'react';
import { ExpensesSummary } from '../../components/expenses-summary';
import { shallow } from 'enzyme';


test('should render the correct snapshot for no expenses', () => {
  const expenseCount = 0;
  const expensesTotal = 0;
  const wrapper = shallow(<ExpensesSummary
  expenseCount={expenseCount}
  expensesTotal={expensesTotal}
/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render the correct snapshot for one expense', () => {
  const expenseCount = 1;
  const expensesTotal = 9434;
  const wrapper = shallow(<ExpensesSummary
  expenseCount={expenseCount}
  expensesTotal={expensesTotal}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('should render the correct snapshot for two expenses', () => {
  const expenseCount = 2;
  const expensesTotal = 9434;
  const wrapper = shallow(<ExpensesSummary
  expenseCount={expenseCount}
  expensesTotal={expensesTotal}
/>);
  expect(wrapper).toMatchSnapshot();
});
