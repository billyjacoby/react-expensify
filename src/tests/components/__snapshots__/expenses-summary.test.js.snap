// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the correct snapshot for no expenses 1`] = `
<div>
  <div>
    <h3>
      No expenses found!
    </h3>
  </div>
</div>
`;

exports[`should render the correct snapshot for one expense 1`] = `
<div>
  <div>
    <h3>
      You are viewing 
      1
       
      expense
       totaling 
      $94.34
    </h3>
  </div>
</div>
`;

exports[`should render the correct snapshot for two expenses 1`] = `
<div>
  <div>
    <h3>
      You are viewing 
      2
       
      expenses
       totaling 
      $94.34
    </h3>
  </div>
</div>
`;
